### ISSUE TITLE
422 on force send

### ISSUE THREAD

**Issue by SSardorf** [sofn-xyz/mailing#322]
**Describe the bug**
I am getting a 422 error in the terminal when I force send.

I am also getting the following error popup:
<img width="441" alt="image" src="https://user-images.githubusercontent.com/22572423/200796345-e02f7ae9-58b7-4c69-a7ff-44ec71b6df68.png">

This is my API route that triggers the sending of the email
```
//api/sendTestEmail.tsx
import sendMail from "emails";
import ConfirmEmail from "emails/ConfirmEmail";

export default async function handler(req: any, res: any) {

  sendMail({
      subject: "First Test Email",
      to: "personalEmail@hidden.com",
      component: <ConfirmEmail firstName="Amelita" ctaText={"Verify this test email"} callbackUrl={"https://callbackUrl.com/"} />
    });
  }
  ```

This is the my emails/index.ts
```
import nodemailer from "nodemailer";
import { buildSendMail } from "mailing-core";

const transport = nodemailer.createTransport({
  pool: true,
  host: "smtp.postmarkapp.com",
  port: 587,
  secure: true, // use TLS
  auth: {
    user: process.env.POSTMARK_CLIENT,
    pass: process.env.POSTMARK_CLIENT,
  },
});

const sendMail = buildSendMail({
  transport,
  defaultFrom: "workEmail@hidden.com",
  configPath: "./mailing.config.json",
});

export default sendMail;

```

Finally, this is the preview output
```
mailing 💌 opening sendMail preview {
  subject: 'First Test Email',
  to: 'personalEmail@hidden.com',
  component: {
    '$$typeof': Symbol(react.element),
    type: [Function: ConfirmEmail],
    key: null,
    ref: null,
    props: {
      firstName: 'Amelita',
      ctaText: 'Verify this test email',
      callbackUrl: 'https://callbackUrl.com/'
    },
    _owner: null,
    _store: {}
  },
  html: 'omitted'
}
API resolved without sending a response for /api/sendTestEmail, this may result in stalled requests.
mailing 💌 opening sendMail preview {
  subject: 'First Test Email',
  to: 'personalEmail@hidden.com',
  component: {
    '$$typeof': Symbol(react.element),
    type: [Function: ConfirmEmail],
    key: null,
    ref: null,
    props: {
      firstName: 'Amelita',
      ctaText: 'Verify this test email',
      callbackUrl: 'https://callbackUrl.com/'
    },
    _owner: null,
    _store: {}
  },
  html: 'omitted'
}
```

**Comment by alexfarrill** [comment]
@SSardorf sorry to hear you are experiencing a problem. Are you using the latest mailing (0.8.15)?

**Comment by SSardorf** [comment]
@alexfarrill Yes I am :)

**Comment by psugihara** [comment]
Dang, sorry for the issue.

My guess is you're hitting getting the 422 in validateApiKey because it's not correctly early returning. The intent here is to protect your mailing server from spammers when it's deployed but open it up in dev.

<img width="1465" alt="Screenshot 2022-11-09 at 9 32 39 AM" src="https://user-images.githubusercontent.com/282016/200900449-35e49ceb-c52b-4d9b-8641-fbf0b0ac2c24.png">

`process.env.NEXT_PUBLIC_MAILING_SKIP_AUTH` gets set here based on your NODE_ENV. Are you running mailing with NODE_ENV="development" or NODE_ENV unset (should default to dev)? Perhaps this condition needs to be refined.

https://github.com/sofn-xyz/mailing/pull/321/files#diff-a20dac20683de447997afdf7e1df78003968af69e9d6a0782a3b92368636b5b7R46

**Comment by elie222** [comment]
I also get the 422 issue

**Comment by psugihara** [comment]
can repro, working on a fix 🏃

**Comment by psugihara** [comment]
fix released in v0.8.16

Thanks @elie222 and @SSardorf for surfacing.

Feel free to re-open and/or ping me in the discord if you're still having problems.
